src/librustc_trait_selection/traits/error_reporting/suggestions.rs

```diff
@@ -760,13 +770,50 @@ impl<'a, 'tcx> InferCtxtExt<'tcx> for InferCtxt<'a, 'tcx> {
             #using-trait-objects-that-allow-for-values-of-different-types>";
         let has_dyn = snippet.split_whitespace().next().map_or(false, |s| s == "dyn");
         let trait_obj = if has_dyn { &snippet[4..] } else { &snippet[..] };
-        if all_returns_have_same_type {
+        if count == 0 {
```

Comment:
Extract branch to a new method. :)

Reply:
on it, I'll have the new commit up after a rebuild `rustc` cycle :)